### ISSUE TITLE
Is pacal calling convention supported?

### ISSUE THREAD

**Issue by rkfg** [NationalSecurityAgency/ghidra#496]
This is related to #485. The Win16 API function calls should use __pascal calling convention for the argument order to be right (left to right). I didn't find anything useful in the code regarding this issue. Is it possible to define a new convention with arguments being pushed left to right? Maybe there's a parameter for a cspec definition that reverses the argument order? So far I couldn't find an easy way to do that.

**Comment by WasserEsser** [comment]
Rightclick -> Edit Function Signature allows you to customize the calling conventions.
You can specify where parameters go, including the stack. 
This can be done left to right and right to left, just set the correct stack locations.

**Comment by rkfg** [comment]
Yeah, I know about this but it's not very convenient. I'd prefer the order to be automatic if possible.

**Comment by GregoryMorse** [comment]
This is a flaw in the Java frontend only, not the back end as pointed out.  The decompiler just expects the stack locations to be correct.  __pascal is __stdcall with a reverse order.  It should be added but also the library definitions need to have knowledge about it for it to be practically useful.

**Comment by GregoryMorse** [comment]
The backend does not really care much about the order.  Its only a display issue.  So the XML would be identical for stdcall.  Underneath the hood the stack offsets are fixed and merely needs to be displayed backwards.  The decompiler does not need to know about it whatsoever.
So exactly how to fix this is less clear - some extra metadata like displayorder="ltr/rtl" could be added again only for the frontend to display it differently.  But perhaps extra metadata in files generally intended only for sleigh/decompiler which will not be used by them is best integrated elsewhere.

**Comment by beholdnec** [comment]
> The backend does not really care much about the order. Its only a display issue. So the XML would be identical for stdcall. Underneath the hood the stack offsets are fixed and merely needs to be displayed backwards.

This is not true in the case of varargs. Functions such as KERNEL:_wsprintf cannot be displayed correctly under this scheme.

**Comment by Wall-AF** [comment]
This is still outstanding and shouldn't be closed.

**Comment by Arcnor** [comment]
So I'm hitting this issue again today. Is there anything on latest Ghidra that can help to make this work? I.e. is the backend now capable of dealing with this and it's just a matter of changing the XML to add the right calling convention?

**Comment by 07AyubHaris** [comment]
I would like to know if this is fixed. I am trying to decompile a 16 bit Windows program and need to use pascal calling convention.

**Comment by Wall-AF** [comment]
@07AyubHaris I've done some work on this on https://github.com/Wall-AF/ghidra/tree/win16/, but in my repo I've messed around locally to the extent that there maybe some subtleties missing!
